cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/reader/ContextConfigurationReader.java

```diff
@@ -216,4 +247,24 @@ public Path getContextRoot() {
 
     return this.contextRoot;
   }
+
+  /**
+   * Search all context.xml configuration files in all subfolders of the template root directory recursively
+   *
+   * @param directory the directory where the context.xml files are searched recursively
+   */
+  private void addContextFilesRecursively(Path directory) {
+
+    Path contextPath = directory.resolve(ConfigurationConstants.CONTEXT_CONFIG_FILENAME);
+    if (Files.exists(contextPath)) {
+      this.contextFiles.add(contextPath);
+    } else {
+      File[] subFolders = directory.toFile().listFiles();
+      for (File subFolder : subFolders) {
+        if (subFolder.isDirectory()) {
+          addContextFilesRecursively(subFolder.toPath());
+        }
+      }
+    }
```

Comment:
this is quite some work and might cause stupid errors in case, someone copies a context.xml for backup to a separate place within the folders. Couldn't we be a little more restrictive on where we would expect context.xml files?